src/metabase/public_settings.clj

```
@@ -205,38 +217,67 @@
-   :setup_token           (resolve-setting 'metabase.setup 'token-value)
-   :site_name             (site-name)
-   :site_url              (site-url)
-   :timezone_short        (short-timezone-name (setting/get :report-timezone))
-   :timezones             common/timezones
-   :types                 (types/types->parents :type/*)
-   :version               config/mb-version-info})
+(defsetting available-locales
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (available-locales-with-names)))
+
+(defsetting email-configured
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.email 'email-configured?)))
+
+(defsetting engines
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (driver.u/available-drivers-info)))
+
+(defsetting entities
+  (deferred-tru "TODO")
+  :visibility :authenticated
+  :getter     (fn [] (types/types->parents :entity/*)))
+
+(defsetting has-sample-dataset
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (db/exists? 'Database, :is_sample true)))
+
+(defsetting hide-embed-branding
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (metastore/hide-embed-branding?)))
+
+(defsetting ldap-configured
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.integrations.ldap 'ldap-configured?)))
+
+(defsetting password-complexity
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] password/active-password-complexity))
+
+(defsetting setup-token
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.setup 'token-value)))
+
+(defsetting timezone-short
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (short-timezone-name (setting/get :report-timezone))))
+
+(defsetting timezones
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] common/timezones))
```

You can't set some of these virtual settings like `:timezones` so they need to have a `:setter` to reflect that. I think the easiest option is to allow `:setter` to be either a function or `:none`. The schema would be 

```clj
(s/cond-pre (s/eq :none) clojure.lang.IFn)
```

Then in whatever function gets the setting (I think it's `metabase.models.setting/set!` if `(= setter :none)` then it should throw an Exception. Something like this is probably fine:

```clj
(when (= setter :none)
  (throw (UnsupportedOperationException. (tru "You cannot set %s; it is a read-only setting." setting-name))))
```